src/test/java/com/commercetools/sync/customobjects/utils/CustomObjectSyncUtilsTest.java

```diff
@@ -0,0 +1,341 @@
+package com.commercetools.sync.customobjects.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.commercetools.sync.commons.utils.SyncUtils.batchElements;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CustomObjectSyncUtilsTest {
+
+    private CustomObject<JsonNode> oldCustomObject;
+    private CustomObjectDraft<JsonNode> newCustomObjectdraft;
+
+    private static final String key = "testkey";
+    private static final String container = "testcontainer";
+
+
+    @Test
+    void hasIdenticalValue_WithSameBooleanValue_ShouldBeIdentical() throws JsonProcessingException {
+        JsonNode actualObj = new ObjectMapper().readTree("true");
+        JsonNode mockedObj = new ObjectMapper().readTree("true");
+
+        newCustomObjectdraft = CustomObjectDraft.ofUnversionedUpsert(container, key, actualObj);
```

Comment:
you could refactor the mocking CustomObject into a private method, also checks like isBoolean, isNumber is not necessary just test the hasIdenticalValue method, but I am also thinking we are doing extra tests which should be part of the Sync class, here it looks like to me just testing the JsonNode which is not necessary to test separately.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/561/commits/249758c5469eb7fe5aa2cab1ad489c4ae987db2e